### DISCUSSION TITLE
I need to obtain the user's IP address, whether it is the front-end or back-end

### DISCUSSION THREAD

**Discussion by tendecade** [nuxt/nuxt#22293] [Questions]
My project needs to obtain the user IP address to limit the form submission limit. At first, it was determined by the backend (JAVA), but now after the project is deployed, the IP addresses obtained by the backend are all 127.0.0.1. Therefore, I need a method to obtain the user IP address, whether it is the front-end or back-end. Thank you～

**Comment by gerkereitz** [comment]
Ideally, you would want do to something like this in the back-end, how are you currently trying to get the user's ip address in your java back-end? Have you looked into existing java rate limiting tools?

If you really wanted to do this in the front-end, which I would not recommend, you could use something like https://www.ipify.org.

P.S. Keep in mind your country's laws and regulations for handling sensitive data and make sure you follow them.
